src/lib.rs

```diff
@@ -920,13 +920,48 @@ impl CertificateParams {
 			});
 		});
 	}
-	fn write_request<K: PublicKeyData>(&self, pub_key: &K, writer :DERWriter) {
+	fn write_request<K: PublicKeyData>(&self, pub_key: &K, writer :DERWriter)
+	 -> Result<(), RcgenError> {
+		// No .. pattern, we use this to ensure every field is used
+		#[deny(unused)]
+		let Self {
+			alg,
+			not_before,
+			not_after,
+			serial_number,
+			subject_alt_names,
+			distinguished_name,
+			is_ca,
+			key_usages,
+			extended_key_usages,
+			name_constraints,
+			custom_extensions,
+			key_pair,
+			use_authority_key_identifier_extension,
+			key_identifier_method,
+		} = self;
+		// - alg and key_pair will be used by the caller
+		// - not_before and not_after cannot be put in a CSR
+		// - There might be a use case for specifying the key identifier
+		// in the CSR, but in the current API it can't be distinguished
+		// from the defaults so this is left for a later version if
+		// needed.
+		let _ = (alg, key_pair, not_before, not_after, key_identifier_method);
+		if !(serial_number.is_none()
```

Comment:
Could you remove this negation by doing de morgan?

Reply:
Sure!